src/main/java/com/flightstats/hub/spoke/SpokeTtlEnforcer.java

```diff
@@ -20,15 +23,24 @@
 @Singleton
 public class SpokeTtlEnforcer {
     private final static Logger logger = LoggerFactory.getLogger(SpokeTtlEnforcer.class);
+    private final SpokeStore spokeStore;
     private final String storagePath;
     private final int ttlMinutes;
+    private long itemsEvicted = 0;
 
     @Inject
     private ChannelService channelService;
 
-    public SpokeTtlEnforcer(String storagePath, int ttlMinutes) {
-        this.storagePath = storagePath;
-        this.ttlMinutes = ttlMinutes + 1;
+    @Inject
+    private MetricsService metricsService;
+
+    @Inject
+    private SpokeContentDao spokeContentDao;
+
+    public SpokeTtlEnforcer(SpokeStore spokeStore) {
+        this.spokeStore = spokeStore;
+        this.storagePath = HubProperties.getSpokePath(spokeStore);
```

Comment:
A longer term thing, but would it make sense to have SpokeStore know it's path and ttl when it's instantiated? Seems like a natural responsibility for SpokeStore to own. Grabbing a spokeStore's info indirectly via HubProperties feels like a responsiblity mismatch.

Reply:
Oh yes, this is mosdef a DI violation.

Reply:
Good catch. I've been drinking the Hub kool-aid long enough that my eyes slid right over that.